kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala

```diff
@@ -407,4 +408,15 @@ object Utils extends Logging {
       stringWriter.toString
     }
   }
+
+  def withLockRequired[T](lock: Lock)(block: => T): T = {
+    var locked: Boolean = false
+    try {
+      lock.lockInterruptibly()
```

Comment:
how do you think about? cc @yaooqinn

Reply:
Also cc @pan3793

Reply:
The state changing can be safely used `lock`, I do not know other places. BTW, how do we handle InteruptExcetion?

Reply:
Reverted